src/openms/source/CONCEPT/Exception.cpp

```diff
@@ -56,42 +57,41 @@ namespace OpenMS
   {
 
     BaseException::BaseException() noexcept :
+      std::runtime_error("unspecified error"),
       file_("?"),
       line_(-1),
       function_("?"),
-      name_("Exception"),
-      what_("unspecified error")
+      name_("Exception")
     {
-      GlobalExceptionHandler::getInstance().set(file_, line_, function_, std::string(name_), std::string(what_));
+      GlobalExceptionHandler::getInstance().set(file_, line_, function_, std::string(name_), std::string(what()));
     }
 
     BaseException::BaseException(const char* file, int line, const char* function, const std::string& name, const std::string& message) noexcept :
+      std::runtime_error(message),
       file_(file),
       line_(line),
       function_(function),
-      name_(name),
-      what_(message)
+      name_(name)
     {
-      GlobalExceptionHandler::getInstance().set(file_, line_, function_, name_, what_);
+      GlobalExceptionHandler::getInstance().set(file_, line_, function_, name_, this->what());
     }
 
     BaseException::BaseException(const char* file, int line, const char* function) noexcept :
+      std::runtime_error("unknown error"),
```

Comment:
Did you specifically want to make a distinction here to the default ctor?

Reply:
no I just used the messages that were there.
I think I can make both unknown error?

Reply:
ah hmm.. don't care much. Maybe it gives a bit more hints where the error came from, this way.